CodeEdit/Features/Tasks/TaskManager.swift

```diff
@@ -59,22 +60,27 @@ class TaskManager: ObservableObject {
     func executeActiveTask() {
         let task = workspaceSettings.tasks.first { $0.id == selectedTaskID }
         guard let task else { return }
-        runTask(task: task)
+        Task {
+            await runTask(task: task)
+        }
     }
 
-    func runTask(task: CETask) {
+    func runTask(task: CETask) async {
         // A process can only be started once, that means we have to renew the Process and Pipe
-        // but don't initialise a new object.
+        // but don't initialize a new object.
         if let activeTask = activeTasks[task.id] {
             activeTask.renew()
+            // Wait until the task is no longer running.
+            // The termination handler is asynchronous, so we avoid a race condition using this.
+            while activeTask.status != .notRunning {
+                await Task.yield()
+            }
```

Comment:
This is causing issue #1838. Any task state other than `.running` represents a finished or not running task
```suggestion
            // Wait until the task is no longer running.
            // The termination handler is asynchronous, so we avoid a race condition using this.
            while activeTask.status == .running {
                await Task.yield()
            }
```

Reply:
Fixed